File: python/demos/demo_mpc.py
#!/usr/bin/env python3
"""MPC demo."""

from __future__ import annotations

from demo_common import run_local

# Imported from the submodule directly, mirroring demo_teb.py's precedent (the
# package re-export lives in navigation/local_planning/predictive/__init__.py).
from navigation.local_planning.predictive.mpc import MpcPlanner

if __name__ == "__main__":
    run_local("mpc", MpcPlanner)

File: cpp/demos/demo_mpc.cpp
#include "demo_common.hpp"
#include "navigation/local_planning/predictive/mpc.hpp"

int main(int argc, char** argv) {
  try {
    return demo::run_local(argc, argv, "mpc", [](const navigation::core::ParamSet& p) {
      return navigation::local_planning::MpcPlanner(p);
    });
  } catch (const std::exception& e) {
    std::cerr << "error: " << e.what() << "\n";
    return 1;
  }
}
